documentloaders/cloudsql/documentloader_option.go

```diff
@@ -0,0 +1,168 @@
+	return func(documentLoader *DocumentLoader) {
+		documentLoader.metadataColumns = metadataColumns
+	}
+}
+
+// WithMetadataJSONColumn sets the column name containing JSON metadata.
+func WithMetadataJSONColumn(metadataJSONColumn string) DocumentLoaderOption {
+	return func(documentLoader *DocumentLoader) {
+		documentLoader.metadataJSONColumn = metadataJSONColumn
+	}
+}
+
+func applyCloudSQLDocumentLoaderOptions(dl *DocumentLoader, options []DocumentLoaderOption) error {
+	for _, opt := range options {
+		opt(dl)
+	}
+
+	return validateFunc(dl)
+}
+
+func validateFunc(dl *DocumentLoader) error {
+	formatters := map[string]func(_ map[string]any, _ []string) string{
+		"csv":  csvFormatter,
+		"":     textFormatter,
+		"text": textFormatter,
+		"json": jsonFormatter,
+		"yaml": yamlFormatter,
+	}
+
+	if dl.engine.Pool == nil {
+		return fmt.Errorf("engine.Pool must be specified")
+	}
+
+	if dl.query == "" && dl.tableName == "" {
+		return fmt.Errorf("either query or tableName must be specified")
+	}
+
+	if dl.query != "" && dl.tableName != "" {
+		return fmt.Errorf("only one of 'table_name' or 'query' should be specified")
+	}
+
+	if dl.format != "" && dl.formatter != nil {
+		return fmt.Errorf("only one of 'format' or 'formatter' must be specified")
+	}
+
+	if dl.query == "" {
+		dl.query = fmt.Sprintf(`SELECT * FROM "%s"."%s"`, dl.schemaName, dl.tableName)
+	}
+
+	if dl.formatter == nil {
+		f, ok := formatters[strings.ToLower(dl.format)]
+		if !ok {
+			return fmt.Errorf("format must be type: 'csv', 'text', 'json', 'yaml'")
+		}
+		dl.formatter = f
+	}
+	return nil
+}
+
+func validateQuery(query string) error {
+	re := regexp.MustCompile(sqlregularexpresion)
+	if !re.MatchString(query) {
+		return fmt.Errorf("query is not valid for the following regular expression: %s", sqlregularexpresion)
+	}
+	return nil
+}
+
+func (l *DocumentLoader) getFieldDescriptions(ctx context.Context) ([]pgconn.FieldDescription, error) {
+	rows, err := l.engine.Pool.Query(ctx, fmt.Sprintf("%s LIMIT 1", l.query))
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute query: %w", err)
+	}
+	defer rows.Close()
+	return rows.FieldDescriptions(), nil
+}
+
+func (l *DocumentLoader) configureColumns(fieldDescriptions []pgconn.FieldDescription) error {
+	if len(l.contentColumns) == 0 {
+		l.contentColumns = []string{fieldDescriptions[0].Name}
+	}
+
+	if len(l.metadataColumns) == 0 {
+		for _, col := range fieldDescriptions {
+			if !slices.Contains(l.contentColumns, col.Name) {
+				l.metadataColumns = append(l.metadataColumns, col.Name)
+			}
+		}
+	}
+
+	if l.metadataJSONColumn == "" {
+		l.metadataJSONColumn = defaultMetadataJSONColumn
+	} else {
+		found := false
+		for _, col := range fieldDescriptions {
+			if col.Name == l.metadataJSONColumn {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return fmt.Errorf("metadata JSON column '%s' not found in query result", l.metadataJSONColumn)
+		}
```

Comment:
It looks like the only field from `fieldDescriptions` we are using is `Name`. 

Can we replace `fieldDescriptions` with a `colNames []string`? That should let us use slices.Contains() here.